### PR TITLE
Enable productionizing frontend by adding CSRF support

### PR DESCRIPTION
**Pull request by aarondav**
## What changes are proposed in this pull request?

In order to help users productionize the mlflow UI, we add CSRF support through an optionally-available cookie, which will be added as a request header to all AJAX API calls, which would otherwise be susceptible to CSRF. This PR does not yet add the support to the flask server, that will come in a follow-up PR.

## How is this patch tested?

- [x] Manual tests - Tested locally.

I was actually unable to do the dev-test cycle using `npm run start`, but that was the case when I started as well. Hopefully someone who reviews this can confirm if that cycle still works for them with these changes.

### CODE REVIEW

File: mlflow/server/templates/compare_runs.html [hunk 1]
```diff
@@ -1,111 +0,0 @@
-{% extends "layout.html" %}
```
Comment (aarondav) [inline mlflow/server/templates/compare_runs.html]: These have been deprecated for a while, just removing them from the source repo.

**Comment by mateiz** [comment]
This is also related to #116, which switched to using fragments in the React router. Which approach do we prefer? The paths look nicer without fragments IMO.

**Comment by aarondav** [comment]
@mateiz Thanks for the comment -- I didn't realize Andrew's work preceded mine. We went ahead and merged his PR, and mine is now rebased on top, only providing CSRF support.

One advantage of the `#` support is that it simplifies the serving requirements; we only need to have one index.html rather than 'autogenerating' them for the specific routes likes `/experiments/0/runs/$run-id`. So I think it's a reasonable way to go forward.

**Comment by mateiz** [comment]
Looks good to me otherwise.

**Comment by aarondav** [comment]
Thanks @mateiz !

**Pull request merged**